openmls/src/group/core_group/validation.rs

```diff
@@ -171,6 +165,59 @@ impl CoreGroup {
                 return Err(ProposalValidationError::InsufficientCapabilities);
             }
 
+            // Leaf node validation.
+            {
+                let leaf_node = add_proposal.add_proposal().key_package().leaf_node();
+
+                let filtered_group_signature_keys = {
```

Comment:
We talked about scoping like this. And this looks like a clear case to me where this should be a function. Then you don't need any of the additional scopes here.